src/android/CameraLauncher.java

```diff
@@ -498,15 +498,19 @@ private void refreshGallery(Uri contentUri)
 
 
 private String ouputModifiedBitmap(Bitmap bitmap, Uri uri) throws IOException {
+        // Some content: URIs do not map to file paths (e.g. picasa).
+        String realPath = FileHelper.getRealPath(uri, this.cordova);
+
+        // Get filename from uri
+        String fileName = realPath != null ? realPath.substring(realPath.lastIndexOf('/') + 1) : "modified.jpg";
+
         // Create an ExifHelper to save the exif data that is lost during compression
-        String modifiedPath = getTempDirectoryPath() + "/modified.jpg";
+        String modifiedPath = getTempDirectoryPath() + "/" + fileName;
 
         OutputStream os = new FileOutputStream(modifiedPath);
         bitmap.compress(Bitmap.CompressFormat.JPEG, this.mQuality, os);
         os.close();
 
-        // Some content: URIs do not map to file paths (e.g. picasa).
-        String realPath = FileHelper.getRealPath(uri, this.cordova);
```

Comment:
These two lines above seems excess, no? You've got `realPath` already at the beginning of this method.